### PR TITLE
fix: [CDS-93678]: add titleContainerClassName to page header

### PR DESCRIPTION
**Pull request by MondalSayantan**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2403

File: packages/uicore/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@harness/uicore",
-  "version": "3.160.5",
+  "version": "3.160.6",
```
Comment (avinashmadhwani02) [inline packages/uicore/package.json:3]: you can increase the minor version since we are adding a new prop and not modifying/fixing anything

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2404

File: packages/uicore/package.json (hunk 1 above)
Comment (MondalSayantan) [reply packages/uicore/package.json]: done

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by PrakharMartand — APPROVED** [review]

**Pull request merged**